File: SuperglideScipt.py
import keyboard
import time
import threading
import hid

# PS5 DualSense Vendor ID and Product ID
VENDOR_ID = 0x054C  # Sony
PRODUCT_ID = 0x0CE6  # DualSense

# Open the PS5 controller
try:
    device = hid.device()
    device.open(VENDOR_ID, PRODUCT_ID)
    print("PS5 Controller connected!")
except Exception as e:
    print(f"Failed to connect to PS5 controller: {e}")
    exit()

# Function to check if a button is pressed
def is_button_pressed():
    # Read input data from the controller
    data = device.read(64)  # Read 64 bytes of data
    if data:
        # Example: Check if L2 (Left Trigger) is pressed
        # Note: You'll need to reverse-engineer the HID report format for the DualSense
        l2_trigger = data[5]  # Example: Byte 5 represents the L2 trigger
        return l2_trigger > 0  # Return True if L2 is pressed
    return False

# Function to spam jump and crouch inputs
def superglide_spam():
    while True:
        if is_button_pressed():  # Only spam inputs while the button is held down
            # Simulate jump (Space) and crouch (Ctrl) inputs
            keyboard.press('space')  # Press jump
            keyboard.press('ctrl')   # Press crouch
            time.sleep(0.01)         # Small delay to simulate simultaneous inputs
            keyboard.release('space')  # Release jump
            keyboard.release('ctrl')   # Release crouch
        else:
            time.sleep(0.01)  # Small delay to avoid high CPU usage

# Start the superglide spam thread
superglide_thread = threading.Thread(target=superglide_spam)
superglide_thread.daemon = True
superglide_thread.start()

# Main loop
print("Hold L2 to spam jump and crouch for supergliding.")
while True:
    time.sleep(0.1)  # Small delay to avoid high CPU usage

File: SuperglideScipt.cpp
#include <windows.h>
#include <Xinput.h>
#include <iostream>
#include <thread>
#include <chrono>

int main() {
    XINPUT_STATE state;
    ZeroMemory(&state, sizeof(XINPUT_STATE));

    bool isHoldingL2 = false;
    auto holdStartTime = std::chrono::steady_clock::now();

    while (true) {
        XInputGetState(0, &state); // Get the current controller state

        // Check if L2 is pressed (Left Trigger)
        if (state.Gamepad.bLeftTrigger > XINPUT_GAMEPAD_TRIGGER_THRESHOLD) {
            if (!isHoldingL2) {
                // Start the hold timer
                isHoldingL2 = true;
                holdStartTime = std::chrono::steady_clock::now();
            } else {
                // Check if L2 has been held for 0.2 seconds
                auto now = std::chrono::steady_clock::now();
                auto holdDuration = std::chrono::duration_cast<std::chrono::milliseconds>(now - holdStartTime).count();

                if (holdDuration >= 200) { // 200ms = 0.2 seconds
                    std::cout << "L2 held for 0.2 seconds. Triggering Left Joystick Click and L2." << std::endl;

                    // Simulate Left Joystick Click (Crouch)
                    state.Gamepad.wButtons |= XINPUT_GAMEPAD_LEFT_THUMB;
                    XInputSetState(0, &state);

                    // Small delay to simulate a "press"
                    std::this_thread::sleep_for(std::chrono::milliseconds(50));

                    // Release Left Joystick Click
                    state.Gamepad.wButtons &= ~XINPUT_GAMEPAD_LEFT_THUMB;
                    XInputSetState(0, &state);

                    // Simulate L2 (Jump)
                    // Note: L2 is a trigger, not a button, so we set the trigger value directly
                    XINPUT_VIBRATION vibration;
                    ZeroMemory(&vibration, sizeof(XINPUT_VIBRATION));
                    vibration.wLeftMotorSpeed = 65535; // Vibrate left motor
                    vibration.wRightMotorSpeed = 65535; // Vibrate right motor
                    XInputSetState(0, &vibration);

                    state.Gamepad.bLeftTrigger = 255; // Max trigger value
                    XInputSetState(0, &state);

                    std::this_thread::sleep_for(std::chrono::milliseconds(50));

                    // Release L2
                    state.Gamepad.bLeftTrigger = 0;
                    XInputSetState(0, &state);
                }
            }
        } else {
            // Reset the hold state if L2 is released
            isHoldingL2 = false;
        }

        std::this_thread::sleep_for(std::chrono::milliseconds(10)); // Small delay to avoid high CPU usage
    }

    return 0;
}
